commands/command_track.go

```diff
@@ -226,6 +230,17 @@ ArgsLoop:
 	err = lockClient.FixFileWriteFlagsInDir(relpath, readOnlyPatterns, writeablePatterns)
 	if err != nil {
 		LoggedError(err, tr.Tr.Get("Error changing lockable file permissions: %s", err))
+		sawError = true
+	}
+
+	if sawError {
+		os.Exit(2)
+	}
+	// If we didn't modify things, but that's because the patterns
+	// were already supported, don't return an error, since what the
+	// user wanted has already been done.
```

Comment:
```suggestion
	// If we didn't modify things, but that's because the patterns
	// were already supported, don't return an error, since what the
	// user wanted has already been done.
	// Otherwise, if we didn't modify things but only because the
	// patterns were disallowed, return an error.
```

Reply:
I've squashed that in with a co-author credit.